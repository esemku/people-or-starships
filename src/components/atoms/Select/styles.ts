import { makeStyles } from '@mui/styles';

export default makeStyles(() => ({
  root: {
    border: '1px solid red',
  },
  inputLabel: {
    textTransform: 'capitalize',
  },
  select: {
    width: 400,
  },
}));
