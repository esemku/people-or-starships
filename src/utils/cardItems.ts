export const PEOPLE_CARD_ITEMS: {
  label: string;
  parameter: string;
}[] = [
  {
    label: 'Name',
    parameter: 'name',
  },
  {
    label: 'Birth Year',
    parameter: 'birth_year',
  },
  {
    label: 'Gender',
    parameter: 'gender',
  },
  {
    label: 'Height',
    parameter: 'height',
  },
  {
    label: 'Eye color',
    parameter: 'eye_color',
  },
];

export const STARSHIPS_CARD_ITEMS: {
  label: string;
  parameter: string;
}[] = [
  {
    label: 'Name',
    parameter: 'name',
  },
  {
    label: 'MGLT',
    parameter: 'MGLT',
  },
  {
    label: 'Model',
    parameter: 'model',
  },
  {
    label: 'Manufacturer',
    parameter: 'manufacturer',
  },
  {
    label: 'Max Atmosphering Speed',
    parameter: 'max_atmosphering_speed',
  },
];
