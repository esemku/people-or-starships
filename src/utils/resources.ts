export const PEOPLE_RESOURCES: {
  label: string;
  name: string;
}[] = [
  {
    label: 'Mass',
    name: 'mass',
  },
  {
    label: 'Height',
    name: 'height',
  },
];

export const STARSHIPS_RESOURCES: {
  label: string;
  name: string;
}[] = [
  {
    label: 'Crew',
    name: 'crew',
  },
  {
    label: 'MGLT',
    name: 'MGLT',
  },
  {
    label: 'Max Atmosphering Speed',
    name: 'max_atmosphering_speed',
  },
  {
    label: 'Passengers',
    name: 'passengers',
  },
];
