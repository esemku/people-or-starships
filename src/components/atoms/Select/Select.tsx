import React, { ChangeEvent } from 'react';
import InputLabel from '@mui/material/InputLabel';
import MenuItem from '@mui/material/MenuItem';
import FormControl from '@mui/material/FormControl';
import Select from '@mui/material/Select';
import { Person } from 'types/people';
import { Starship } from 'types/starships';
import { PEOPLE_RESOURCES, STARSHIPS_RESOURCES } from 'utils/resources';
import { nanoid } from 'nanoid';
import useStyles from './styles';

type SelectProps = {
  options: typeof PEOPLE_RESOURCES | typeof STARSHIPS_RESOURCES;
  onChange: (value: keyof Person | keyof Starship) => void;
  value: keyof Person | keyof Starship;
};

const FormSelect: React.FC<SelectProps> = ({
  options,
  onChange,
  value,
}: SelectProps) => {
  const styles = useStyles();

  return (
    <FormControl>
      <InputLabel id="demo-simple-select-label" className={styles.inputLabel}>
        {value}
      </InputLabel>
      <Select
        labelId="demo-simple-select-label"
        id="demo-simple-select"
        value={value}
        label={value}
        onChange={(e: ChangeEvent<HTMLInputElement>) =>
          onChange(e.target.value as keyof Person | keyof Starship)
        }
        className={styles.select}
      >
        {options.map((option) => (
          <MenuItem value={option.name} key={nanoid()}>
            {option.label}
          </MenuItem>
        ))}
      </Select>
    </FormControl>
  );
};

export default FormSelect;
