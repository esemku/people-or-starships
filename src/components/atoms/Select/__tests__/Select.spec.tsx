import React from 'react';
import { render, screen } from '@testing-library/react';

import { PEOPLE_RESOURCES } from 'utils/resources';
import Select from '../Select';

describe('<Select />', () => {
  it('renders correctly', () => {
    const { container } = render(
      <Select options={PEOPLE_RESOURCES} onChange={jest.fn} value="mass" />,
    );

    expect(container).toMatchSnapshot();
  });
});
