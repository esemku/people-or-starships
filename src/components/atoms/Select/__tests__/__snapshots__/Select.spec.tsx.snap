// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Select /> renders correctly 1`] = `
<div>
  <div
    class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
  >
    <label
      class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled makeStyles-inputLabel-2 css-1kty9di-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="true"
      id="demo-simple-select-label"
    >
      mass
    </label>
    <div
      class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl makeStyles-select-3 css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
    >
      <div
        aria-expanded="false"
        aria-haspopup="listbox"
        aria-labelledby="demo-simple-select-label demo-simple-select"
        class="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
        id="demo-simple-select"
        role="button"
        tabindex="0"
      >
        Mass
      </div>
      <input
        aria-hidden="true"
        class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
        tabindex="-1"
        value="mass"
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
        data-testid="ArrowDropDownIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-1z7n62"
        >
          <span>
            mass
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</div>
`;
